test/client_test.dart

```diff
@@ -46,31 +91,31 @@ void main() {
       );
       final data = response.data;
       final error = response.error;
-      expect(error, isNull);
-      expect(data?.accessToken is String, true);
-      expect(data?.refreshToken is String, true);
-      expect(data?.user?.id is String, true);
+      expect(error?.message, isNull);
```

Comment:
why only the message is null?

Reply:
It isn't because only the message is null. It's because if the error isn't null, it's nice to have the message in the log output. Before, when error wasn't null, the test fails and outputs:
```
Expected: null
Actual: <Instance of 'GotrueError'>
```

With the new code, it will show:
```
Expected: null
Actual: Error message appears here
```

Granted, this would create a loophole if the error were not null but the message was null. However, in this case, GoTrueError object does not allow a null error message. So we are safe with this implementation and it is more verbose on failure.

Reply:
why don't check if the code is null, if possible?

Reply:
Because we don't have a way to check if it is null because it was supplied as null vs null because it wasn't supplied.

Reply:
Personally really like this improvement! Makes debugging a lot easier.

Reply:
Apologies, I mistook this for the other review chain. I have edited my previous response and also copied it here:

@bdlukaa not sure I follow. Are you talking about the status code? Which code are you wanting to check / what alternative test would you propose and why?